src/avro_util.erl

```diff
@@ -295,6 +296,127 @@ resolve_duplicated_refs(Type0) ->
   {Type, _Refs} = resolve_duplicated_refs(Type0, []),
   Type.
 
+%% @doc Checks whether two schemas are compatible in sense that data
+%% encoded with the writer schema can be decoded by the reader schema.
+%% Schema compatibility is described here:
+%% https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution
+%% An exception in this implementation is the comparison of two unions.
+%% Check considers union member order changes to be incompatible. This
+%% means that in order for a writer union to be compatible with the
+%% reader union, the writer union must only add new members to it.
+%% For example
+%% is_compatible([int, string], [int, string]) = TRUE
+%% is_compatible([string, int], [int, string]) = FALSE
+%% is_compatible([string, int], [string]) = TRUE
+%% is_compatible([string, int], [int]) = FALSE
+%% is_compatible([int, string], [int, string, some_record]) = TRUE
+%% @end
+-spec is_compatible(avro_type(), avro_type()) -> true | {not_compatible, _, _}.
+is_compatible(Reader, Writer) ->
+  try
+    do_is_compatible(Reader, Writer, [], [])
+  catch throw:{not_compatible, RPath, WPath} ->
+      {not_compatible, lists:reverse(RPath), lists:reverse(WPath)};
+        throw:{reader_missing_defalut_value, Path} ->
+      {reader_missing_default_value, lists:reverse(Path)}
+  end.
+
+do_is_compatible(Reader, Writer, RPath, WPath) ->
+  NewRPath = [avro:get_type_fullname(Reader) | RPath],
+  NewWPath = [avro:get_type_fullname(Writer) | WPath],
+  do_is_compatible(Reader, Writer, RPath, WPath, NewRPath, NewWPath).
```

Comment:
there is no need to pass `RPath` and `WPath` down the stack because
`RPath = tl(NewRPath)` and `WPath = tl(NewWPath)`